packages/grafana-ui/src/components/Text/Text.tsx

```diff
@@ -0,0 +1,130 @@
+import { css, cx } from '@emotion/css';
+import React, { createElement, CSSProperties, useCallback } from 'react';
+
+import { GrafanaTheme2, ThemeTypographyVariantTypes } from '@grafana/data';
+
+import { useStyles2 } from '../../themes';
+
+interface TextProps {
+  /** Defines what HTML element is defined underneath, also maps into a default typography variant */
+  as: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'span' | 'p' | 'legend';
+  /** What typograpy variant should be used for the component. Only use if default variant for the defined 'as' is not what is needed */
+  variant?: keyof ThemeTypographyVariantTypes;
+  /** Override the default weight for the used variant */
+  weight?: 'light' | 'regular' | 'medium' | 'bold';
+  /** Color to use for text */
+  color?: keyof GrafanaTheme2['colors']['text'] | 'error' | 'success' | 'warning' | 'info';
+  /** Use to cut the text off with ellipsis if there isn't space to show all of it. On hover shows the rest of the text */
+  truncate?: boolean;
+  /** Whether to align the text to left, center or right */
+  textAlignment?: CSSProperties['textAlign'];
+  /** Margin to set on the component. Remember, it does not work with inline elements such as 'span'. */
+  margin?: string | number;
```

Comment:
I think it's fine to keep margin props, with the changes I suggested, and docs encouraging using `<Stack>` instead (like for stacking multiple paragraphs, rather than setting marginBottom each time)

Reply:
I 1000% agree with what you said. The fact we don't have a `Stack` component in grafana-ui threw me off thinking we could do something like what you said in the future, but instead this should just get us to get that `Stack` component in grafana-ui quicker.

_Ideally_ what we would have in the end is exactly what you described in the first point, a proper spacing token structure that we then can just do `margin='1'` and it sets it to whatever `spacing-1` token is. We aren't there though 😭  I'm of two minds about this:
1) We keep margin, add some sort of "spacing tokens" like you said, knowingly running the risk of our actual space tokens not looking like this in the future and we have to make breaking changes to this margin prop
2) We don't add margin right now until we make a decision about spacing tokens and then we can add more consistent margin props to components.

If we think we can successfully do 1 and then in the future change it I think we should go for it.

Regarding only having `marginTop` and `marginBottom`, I am less certain. We could encourage people through documentation to only set vertical margin on text and use Stack anywhere else, but unfortunately from my exploration there are quite a few places that set margins in every direction 😞

Reply:
This is the case for `@grafana/ui/experimental` export. What I had in mind:

1. Make up some space scales we think would be appropriate. We use our best judgement here but don't spend too much time antagonising an just go for it.
2. Put component in `@grafana/ui/experimental` and start converting a bunch of grafana core over to use it to prove our that we made the right decisions
   - If we find our space scale doesn't work, we change it. 
4. Make `<Stack />`. Stack would use the same space scale, put it in `@grafana/ui/experimental`, start testing in grafana core to validate the space scale. Tweak if necessary.
5. Take both components out of experimental.

> Regarding only having `marginTop` and `marginBottom`, I am less certain.

I don't see the use case for marginLeft/Right on text/headings that isn't better served by Stack.

Reply:
I think Stack should have a grid unit spacing/gap prop (like it has now). Makes it a lot more flexible. But do miss named fontSizes, sm, md, lg, etc (Which we don't have in the theme).

Reply:
What is margin prop doing? Setting bottom margin in grid units? I think that could be useful, but I don't see any other need for margin prop (other than overring the default h1-6 bottom margin).

Reply:
@JoaoSilvaGrafana I think the default h1-6 styles only have bottom margin.....after checking it appears they have no bottom margin in the theme, the bottom margin comes from the old sass styles. 

I am not sure, honestly. 

* but having a margin that is a string prop is bad I think (can be set to non grid units) 
* Having a margin that only change margin in all directions (up, right, down, left) feels unusable as I can't think of many scenarios where you would want uniform margins in all directions. 

But I would expect some built-in default bottom margin/spacing for h1-4, what does other component libraries do?

Reply:
In terms of having default margins, most of them seem to not have it and in their examples just use some sort of Stack component, but some do have them, like [Adobe](https://spectrum.adobe.com/page/heading/#Margins) and [Github](https://primer.style/react/Text). Feels like a design decision on how we would want to represent these and they should match figma.

Out of the DS I checked a lot of them did not have a margin prop, some that did:
- [DataDog](https://druids.datadoghq.com/components/typography/Text#margins) has some `SPACING_SIZE` options, which I am assuming relates to their [tokens](https://druids.datadoghq.com/foundations/spacing-and-layout) 
- [Kiwi.com](https://orbit.kiwi/components/text/text/react/)  seems to allow string or numbers, like we currently do in this PR
- [Grommet](https://v2.grommet.io/text#margin) only allows token-like strings

So I think our possible options would be to only allow margin props that relate to our tokens or to not allow them at all and point people to our future layout tokens when they exist.

Reply:
As we moved the `Text component` to the **unstable** package with the objective of getting real feedback about it, we will know if a `margin` prop is missed by the devs using the `Text component`. 
So from my honest point of view, I will go for resetting `margin` and `padding` and not having a `margin` prop.